Generated classes create conflicts with InternalsVisibleTo

Using the 3.0 b2 msbuild tasks, I see this:
```
C:\Users\Oren\Source\Git\MetroLog\MetroLog.WP8\obj\Release\GitVersionTaskAssemblyInfo.g.cs(8,12,8,23): warning CS0436: The type 'ReleaseDateAttribute' in 'C:\Users\Oren\Source\Git\MetroLog\MetroLog.WP8\obj\Release\GitVersionTaskAssemblyInfo.g.cs' conflicts with the imported type 'ReleaseDateAttribute' in 'MetroLog, Version=0.8.8.0, Culture=neutral, PublicKeyToken=ba4ace74c3b410f3'. Using the type defined in 'C:\Users\Oren\Source\Git\MetroLog\MetroLog.WP8\obj\Release\GitVersionTaskAssemblyInfo.g.cs'.
```
Problem is that the ReleaseDateAttribute and GitVersionInformation classes are created in the root namespace, so they'll conflict with friend access.

Proposal:
Put the attribute in a namespace underneath the assembly name.

I.e., for an assemlby Foobar.Frob:

namespace Foobar.Frob {
...
}

--- a/.gitattributes
+++ b/.gitattributes
@@ -21,5 +21,3 @@
 *.snk binary
 *.pdb binary
 *.svg binary
-
-*.approved.* binary

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_Major.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_Major.approved.txt
@@ -5,22 +5,24 @@ using System.Reflection;
 [assembly: AssemblyVersion("2.0.0.0")]
 [assembly: AssemblyFileVersion("2.3.4.0")]
 [assembly: AssemblyInformationalVersion("2.3.4-beta.5+6.Branch.master.Sha.commitSha")]
-[assembly: ReleaseDate("2014-03-06")]
+[assembly: Fake.ReleaseDate("2014-03-06")]
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-sealed class ReleaseDateAttribute : System.Attribute
+namespace Fake
 {
-    public string Date { get; private set; }
-
-    public ReleaseDateAttribute(string date)
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    sealed class ReleaseDateAttribute : System.Attribute
     {
-        Date = date;
-    }
-}
+        public string Date { get; private set; }
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-static class GitVersionInformation
-{
+        public ReleaseDateAttribute(string date)
+        {
+            Date = date;
+        }
+    }
+
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    static class GitVersionInformation
+    {
     public static string Major = "2";
     public static string Minor = "3";
     public static string Patch = "4";
@@ -41,6 +43,5 @@ static class GitVersionInformation
     public static string NuGetVersion = "2.3.4-beta0005";
     public static string CommitDate = "2014-03-06";
 
+    }
 }
-
-

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_Major.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_Major.approved.txt
@@ -23,25 +23,25 @@ namespace Fake
     [System.Runtime.CompilerServices.CompilerGenerated]
     static class GitVersionInformation
     {
-    public static string Major = "2";
-    public static string Minor = "3";
-    public static string Patch = "4";
-    public static string PreReleaseTag = "beta.5";
-    public static string PreReleaseTagWithDash = "-beta.5";
-    public static string BuildMetaData = "6";
-    public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
-    public static string MajorMinorPatch = "2.3.4";
-    public static string SemVer = "2.3.4-beta.5";
-    public static string LegacySemVer = "2.3.4-beta5";
-    public static string LegacySemVerPadded = "2.3.4-beta0005";
-    public static string AssemblySemVer = "2.0.0.0";
-    public static string FullSemVer = "2.3.4-beta.5+6";
-    public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
-    public static string BranchName = "master";
-    public static string Sha = "commitSha";
-    public static string NuGetVersionV2 = "2.3.4-beta0005";
-    public static string NuGetVersion = "2.3.4-beta0005";
-    public static string CommitDate = "2014-03-06";
+        public static string Major = "2";
+        public static string Minor = "3";
+        public static string Patch = "4";
+        public static string PreReleaseTag = "beta.5";
+        public static string PreReleaseTagWithDash = "-beta.5";
+        public static string BuildMetaData = "6";
+        public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
+        public static string MajorMinorPatch = "2.3.4";
+        public static string SemVer = "2.3.4-beta.5";
+        public static string LegacySemVer = "2.3.4-beta5";
+        public static string LegacySemVerPadded = "2.3.4-beta0005";
+        public static string AssemblySemVer = "2.0.0.0";
+        public static string FullSemVer = "2.3.4-beta.5+6";
+        public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
+        public static string BranchName = "master";
+        public static string Sha = "commitSha";
+        public static string NuGetVersionV2 = "2.3.4-beta0005";
+        public static string NuGetVersion = "2.3.4-beta0005";
+        public static string CommitDate = "2014-03-06";
 
     }
 }

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinor.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinor.approved.txt
@@ -5,22 +5,24 @@ using System.Reflection;
 [assembly: AssemblyVersion("2.3.0.0")]
 [assembly: AssemblyFileVersion("2.3.4.0")]
 [assembly: AssemblyInformationalVersion("2.3.4-beta.5+6.Branch.master.Sha.commitSha")]
-[assembly: ReleaseDate("2014-03-06")]
+[assembly: Fake.ReleaseDate("2014-03-06")]
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-sealed class ReleaseDateAttribute : System.Attribute
+namespace Fake
 {
-    public string Date { get; private set; }
-
-    public ReleaseDateAttribute(string date)
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    sealed class ReleaseDateAttribute : System.Attribute
     {
-        Date = date;
-    }
-}
+        public string Date { get; private set; }
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-static class GitVersionInformation
-{
+        public ReleaseDateAttribute(string date)
+        {
+            Date = date;
+        }
+    }
+
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    static class GitVersionInformation
+    {
     public static string Major = "2";
     public static string Minor = "3";
     public static string Patch = "4";
@@ -41,6 +43,5 @@ static class GitVersionInformation
     public static string NuGetVersion = "2.3.4-beta0005";
     public static string CommitDate = "2014-03-06";
 
+    }
 }
-
-

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinor.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinor.approved.txt
@@ -23,25 +23,25 @@ namespace Fake
     [System.Runtime.CompilerServices.CompilerGenerated]
     static class GitVersionInformation
     {
-    public static string Major = "2";
-    public static string Minor = "3";
-    public static string Patch = "4";
-    public static string PreReleaseTag = "beta.5";
-    public static string PreReleaseTagWithDash = "-beta.5";
-    public static string BuildMetaData = "6";
-    public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
-    public static string MajorMinorPatch = "2.3.4";
-    public static string SemVer = "2.3.4-beta.5";
-    public static string LegacySemVer = "2.3.4-beta5";
-    public static string LegacySemVerPadded = "2.3.4-beta0005";
-    public static string AssemblySemVer = "2.3.0.0";
-    public static string FullSemVer = "2.3.4-beta.5+6";
-    public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
-    public static string BranchName = "master";
-    public static string Sha = "commitSha";
-    public static string NuGetVersionV2 = "2.3.4-beta0005";
-    public static string NuGetVersion = "2.3.4-beta0005";
-    public static string CommitDate = "2014-03-06";
+        public static string Major = "2";
+        public static string Minor = "3";
+        public static string Patch = "4";
+        public static string PreReleaseTag = "beta.5";
+        public static string PreReleaseTagWithDash = "-beta.5";
+        public static string BuildMetaData = "6";
+        public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
+        public static string MajorMinorPatch = "2.3.4";
+        public static string SemVer = "2.3.4-beta.5";
+        public static string LegacySemVer = "2.3.4-beta5";
+        public static string LegacySemVerPadded = "2.3.4-beta0005";
+        public static string AssemblySemVer = "2.3.0.0";
+        public static string FullSemVer = "2.3.4-beta.5+6";
+        public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
+        public static string BranchName = "master";
+        public static string Sha = "commitSha";
+        public static string NuGetVersionV2 = "2.3.4-beta0005";
+        public static string NuGetVersion = "2.3.4-beta0005";
+        public static string CommitDate = "2014-03-06";
 
     }
 }

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatch.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatch.approved.txt
@@ -23,25 +23,25 @@ namespace Fake
     [System.Runtime.CompilerServices.CompilerGenerated]
     static class GitVersionInformation
     {
-    public static string Major = "2";
-    public static string Minor = "3";
-    public static string Patch = "4";
-    public static string PreReleaseTag = "beta.5";
-    public static string PreReleaseTagWithDash = "-beta.5";
-    public static string BuildMetaData = "6";
-    public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
-    public static string MajorMinorPatch = "2.3.4";
-    public static string SemVer = "2.3.4-beta.5";
-    public static string LegacySemVer = "2.3.4-beta5";
-    public static string LegacySemVerPadded = "2.3.4-beta0005";
-    public static string AssemblySemVer = "2.3.4.0";
-    public static string FullSemVer = "2.3.4-beta.5+6";
-    public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
-    public static string BranchName = "master";
-    public static string Sha = "commitSha";
-    public static string NuGetVersionV2 = "2.3.4-beta0005";
-    public static string NuGetVersion = "2.3.4-beta0005";
-    public static string CommitDate = "2014-03-06";
+        public static string Major = "2";
+        public static string Minor = "3";
+        public static string Patch = "4";
+        public static string PreReleaseTag = "beta.5";
+        public static string PreReleaseTagWithDash = "-beta.5";
+        public static string BuildMetaData = "6";
+        public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
+        public static string MajorMinorPatch = "2.3.4";
+        public static string SemVer = "2.3.4-beta.5";
+        public static string LegacySemVer = "2.3.4-beta5";
+        public static string LegacySemVerPadded = "2.3.4-beta0005";
+        public static string AssemblySemVer = "2.3.4.0";
+        public static string FullSemVer = "2.3.4-beta.5+6";
+        public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
+        public static string BranchName = "master";
+        public static string Sha = "commitSha";
+        public static string NuGetVersionV2 = "2.3.4-beta0005";
+        public static string NuGetVersion = "2.3.4-beta0005";
+        public static string CommitDate = "2014-03-06";
 
     }
 }

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatch.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatch.approved.txt
@@ -5,22 +5,24 @@ using System.Reflection;
 [assembly: AssemblyVersion("2.3.4.0")]
 [assembly: AssemblyFileVersion("2.3.4.0")]
 [assembly: AssemblyInformationalVersion("2.3.4-beta.5+6.Branch.master.Sha.commitSha")]
-[assembly: ReleaseDate("2014-03-06")]
+[assembly: Fake.ReleaseDate("2014-03-06")]
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-sealed class ReleaseDateAttribute : System.Attribute
+namespace Fake
 {
-    public string Date { get; private set; }
-
-    public ReleaseDateAttribute(string date)
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    sealed class ReleaseDateAttribute : System.Attribute
     {
-        Date = date;
-    }
-}
+        public string Date { get; private set; }
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-static class GitVersionInformation
-{
+        public ReleaseDateAttribute(string date)
+        {
+            Date = date;
+        }
+    }
+
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    static class GitVersionInformation
+    {
     public static string Major = "2";
     public static string Minor = "3";
     public static string Patch = "4";
@@ -41,6 +43,5 @@ static class GitVersionInformation
     public static string NuGetVersion = "2.3.4-beta0005";
     public static string CommitDate = "2014-03-06";
 
+    }
 }
-
-

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatchTag.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatchTag.approved.txt
@@ -5,22 +5,24 @@ using System.Reflection;
 [assembly: AssemblyVersion("2.3.4.5")]
 [assembly: AssemblyFileVersion("2.3.4.0")]
 [assembly: AssemblyInformationalVersion("2.3.4-beta.5+6.Branch.master.Sha.commitSha")]
-[assembly: ReleaseDate("2014-03-06")]
+[assembly: Fake.ReleaseDate("2014-03-06")]
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-sealed class ReleaseDateAttribute : System.Attribute
+namespace Fake
 {
-    public string Date { get; private set; }
-
-    public ReleaseDateAttribute(string date)
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    sealed class ReleaseDateAttribute : System.Attribute
     {
-        Date = date;
-    }
-}
+        public string Date { get; private set; }
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-static class GitVersionInformation
-{
+        public ReleaseDateAttribute(string date)
+        {
+            Date = date;
+        }
+    }
+
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    static class GitVersionInformation
+    {
     public static string Major = "2";
     public static string Minor = "3";
     public static string Patch = "4";
@@ -41,6 +43,5 @@ static class GitVersionInformation
     public static string NuGetVersion = "2.3.4-beta0005";
     public static string CommitDate = "2014-03-06";
 
+    }
 }
-
-

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatchTag.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyAssemblyVersion_MajorMinorPatchTag.approved.txt
@@ -23,25 +23,25 @@ namespace Fake
     [System.Runtime.CompilerServices.CompilerGenerated]
     static class GitVersionInformation
     {
-    public static string Major = "2";
-    public static string Minor = "3";
-    public static string Patch = "4";
-    public static string PreReleaseTag = "beta.5";
-    public static string PreReleaseTagWithDash = "-beta.5";
-    public static string BuildMetaData = "6";
-    public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
-    public static string MajorMinorPatch = "2.3.4";
-    public static string SemVer = "2.3.4-beta.5";
-    public static string LegacySemVer = "2.3.4-beta5";
-    public static string LegacySemVerPadded = "2.3.4-beta0005";
-    public static string AssemblySemVer = "2.3.4.5";
-    public static string FullSemVer = "2.3.4-beta.5+6";
-    public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
-    public static string BranchName = "master";
-    public static string Sha = "commitSha";
-    public static string NuGetVersionV2 = "2.3.4-beta0005";
-    public static string NuGetVersion = "2.3.4-beta0005";
-    public static string CommitDate = "2014-03-06";
+        public static string Major = "2";
+        public static string Minor = "3";
+        public static string Patch = "4";
+        public static string PreReleaseTag = "beta.5";
+        public static string PreReleaseTagWithDash = "-beta.5";
+        public static string BuildMetaData = "6";
+        public static string FullBuildMetaData = "6.Branch.master.Sha.commitSha";
+        public static string MajorMinorPatch = "2.3.4";
+        public static string SemVer = "2.3.4-beta.5";
+        public static string LegacySemVer = "2.3.4-beta5";
+        public static string LegacySemVerPadded = "2.3.4-beta0005";
+        public static string AssemblySemVer = "2.3.4.5";
+        public static string FullSemVer = "2.3.4-beta.5+6";
+        public static string InformationalVersion = "2.3.4-beta.5+6.Branch.master.Sha.commitSha";
+        public static string BranchName = "master";
+        public static string Sha = "commitSha";
+        public static string NuGetVersionV2 = "2.3.4-beta0005";
+        public static string NuGetVersion = "2.3.4-beta0005";
+        public static string CommitDate = "2014-03-06";
 
     }
 }

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyCreatedCode.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyCreatedCode.approved.txt
@@ -23,25 +23,25 @@ namespace Fake
     [System.Runtime.CompilerServices.CompilerGenerated]
     static class GitVersionInformation
     {
-    public static string Major = "1";
-    public static string Minor = "2";
-    public static string Patch = "3";
-    public static string PreReleaseTag = "unstable.4";
-    public static string PreReleaseTagWithDash = "-unstable.4";
-    public static string BuildMetaData = "5";
-    public static string FullBuildMetaData = "5.Branch.feature1.Sha.commitSha";
-    public static string MajorMinorPatch = "1.2.3";
-    public static string SemVer = "1.2.3-unstable.4";
-    public static string LegacySemVer = "1.2.3-unstable4";
-    public static string LegacySemVerPadded = "1.2.3-unstable0004";
-    public static string AssemblySemVer = "1.2.3.0";
-    public static string FullSemVer = "1.2.3-unstable.4+5";
-    public static string InformationalVersion = "1.2.3-unstable.4+5.Branch.feature1.Sha.commitSha";
-    public static string BranchName = "feature1";
-    public static string Sha = "commitSha";
-    public static string NuGetVersionV2 = "1.2.3-unstable0004";
-    public static string NuGetVersion = "1.2.3-unstable0004";
-    public static string CommitDate = "2014-03-06";
+        public static string Major = "1";
+        public static string Minor = "2";
+        public static string Patch = "3";
+        public static string PreReleaseTag = "unstable.4";
+        public static string PreReleaseTagWithDash = "-unstable.4";
+        public static string BuildMetaData = "5";
+        public static string FullBuildMetaData = "5.Branch.feature1.Sha.commitSha";
+        public static string MajorMinorPatch = "1.2.3";
+        public static string SemVer = "1.2.3-unstable.4";
+        public static string LegacySemVer = "1.2.3-unstable4";
+        public static string LegacySemVerPadded = "1.2.3-unstable0004";
+        public static string AssemblySemVer = "1.2.3.0";
+        public static string FullSemVer = "1.2.3-unstable.4+5";
+        public static string InformationalVersion = "1.2.3-unstable.4+5.Branch.feature1.Sha.commitSha";
+        public static string BranchName = "feature1";
+        public static string Sha = "commitSha";
+        public static string NuGetVersionV2 = "1.2.3-unstable0004";
+        public static string NuGetVersion = "1.2.3-unstable0004";
+        public static string CommitDate = "2014-03-06";
 
     }
 }

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyCreatedCode.approved.txt
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.VerifyCreatedCode.approved.txt
@@ -5,22 +5,24 @@ using System.Reflection;
 [assembly: AssemblyVersion("1.2.3.0")]
 [assembly: AssemblyFileVersion("1.2.3.0")]
 [assembly: AssemblyInformationalVersion("1.2.3-unstable.4+5.Branch.feature1.Sha.commitSha")]
-[assembly: ReleaseDate("2014-03-06")]
+[assembly: Fake.ReleaseDate("2014-03-06")]
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-sealed class ReleaseDateAttribute : System.Attribute
+namespace Fake
 {
-    public string Date { get; private set; }
-
-    public ReleaseDateAttribute(string date)
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    sealed class ReleaseDateAttribute : System.Attribute
     {
-        Date = date;
-    }
-}
+        public string Date { get; private set; }
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-static class GitVersionInformation
-{
+        public ReleaseDateAttribute(string date)
+        {
+            Date = date;
+        }
+    }
+
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    static class GitVersionInformation
+    {
     public static string Major = "1";
     public static string Minor = "2";
     public static string Patch = "3";
@@ -41,6 +43,5 @@ static class GitVersionInformation
     public static string NuGetVersion = "1.2.3-unstable0004";
     public static string CommitDate = "2014-03-06";
 
+    }
 }
-
-

--- a/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.cs
+++ b/src/GitVersionTask.Tests/AssemblyInfoBuilderTests.cs
@@ -11,7 +11,6 @@ using NUnit.Framework;
 [TestFixture]
 public class AssemblyInfoBuilderTests
 {
-
     [Test]
     public void VerifyCreatedCode()
     {
@@ -26,14 +25,13 @@ public class AssemblyInfoBuilderTests
         };
         var assemblyInfoBuilder = new AssemblyInfoBuilder();
         var versionVariables = VariableProvider.GetVariablesFor(semanticVersion, AssemblyVersioningScheme.MajorMinorPatch, VersioningMode.ContinuousDelivery, "ci", false);
-        var assemblyInfoText = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables);
+        var assemblyInfoText = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables, "Fake");
         Approvals.Verify(assemblyInfoText);
 
         var compilation = CSharpCompilation.Create("Fake.dll")
             .WithOptions(new CSharpCompilationOptions(OutputKind.NetModule))
             .AddReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location))
             .AddSyntaxTrees(CSharpSyntaxTree.ParseText(assemblyInfoText));
-        
 
         var emitResult = compilation.Emit(new MemoryStream());
         Assert.IsTrue(emitResult.Success, string.Join(Environment.NewLine, emitResult.Diagnostics.Select(x => x.Descriptor)));
@@ -80,7 +78,7 @@ public class AssemblyInfoBuilderTests
         };
         var assemblyInfoBuilder = new AssemblyInfoBuilder();
         var versionVariables = VariableProvider.GetVariablesFor(semanticVersion, avs, VersioningMode.ContinuousDelivery, "ci", false);
-        var assemblyInfoText = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables);
+        var assemblyInfoText = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables, "Fake");
         Approvals.Verify(assemblyInfoText);
 
         var compilation = CSharpCompilation.Create("Fake.dll")

--- a/src/GitVersionTask/AssemblyInfoBuilder/AssemblyInfoBuilder.cs
+++ b/src/GitVersionTask/AssemblyInfoBuilder/AssemblyInfoBuilder.cs
@@ -50,7 +50,7 @@ namespace {5}
         var members = new StringBuilder();
         foreach (var variable in vars)
         {
-            members.AppendLine(string.Format("    public static string {0} = \"{1}\";", variable.Key, variable.Value));
+            members.AppendLine(string.Format("        public static string {0} = \"{1}\";", variable.Key, variable.Value));
         }
 
         return members.ToString();

--- a/src/GitVersionTask/AssemblyInfoBuilder/AssemblyInfoBuilder.cs
+++ b/src/GitVersionTask/AssemblyInfoBuilder/AssemblyInfoBuilder.cs
@@ -4,7 +4,7 @@ using GitVersion;
 
 public class AssemblyInfoBuilder
 {
-    public string GetAssemblyInfoText(VersionVariables vars)
+    public string GetAssemblyInfoText(VersionVariables vars, string assemblyName)
     {
         var assemblyInfo = string.Format(@"
 using System;
@@ -13,32 +13,34 @@ using System.Reflection;
 [assembly: AssemblyVersion(""{0}"")]
 [assembly: AssemblyFileVersion(""{1}"")]
 [assembly: AssemblyInformationalVersion(""{2}"")]
-[assembly: ReleaseDate(""{3}"")]
+[assembly: {5}.ReleaseDate(""{3}"")]
 
-[System.Runtime.CompilerServices.CompilerGenerated]
-sealed class ReleaseDateAttribute : System.Attribute
+namespace {5}
 {{
-    public string Date {{ get; private set; }}
-
-    public ReleaseDateAttribute(string date)
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    sealed class ReleaseDateAttribute : System.Attribute
     {{
-        Date = date;
+        public string Date {{ get; private set; }}
+
+        public ReleaseDateAttribute(string date)
+        {{
+            Date = date;
+        }}
+    }}
+
+    [System.Runtime.CompilerServices.CompilerGenerated]
+    static class GitVersionInformation
+    {{
+{4}
     }}
 }}
-
-[System.Runtime.CompilerServices.CompilerGenerated]
-static class GitVersionInformation
-{{
-{4}
-}}
-
-
-", 
-vars.AssemblySemVer,
- vars.MajorMinorPatch + ".0", 
- vars.InformationalVersion,
-            vars.CommitDate,
-            GenerateVariableMembers(vars));
+",
+        vars.AssemblySemVer,
+        vars.MajorMinorPatch + ".0",
+        vars.InformationalVersion,
+        vars.CommitDate,
+        GenerateVariableMembers(vars),
+        assemblyName);
 
         return assemblyInfo;
     }
@@ -53,5 +55,4 @@ vars.AssemblySemVer,
 
         return members.ToString();
     }
-
 }

--- a/src/GitVersionTask/AssemblyInfoBuilder/UpdateAssemblyInfo.cs
+++ b/src/GitVersionTask/AssemblyInfoBuilder/UpdateAssemblyInfo.cs
@@ -69,7 +69,7 @@
             TempFileTracker.DeleteTempFiles();
 
             InvalidFileChecker.CheckForInvalidFiles(CompileFiles, ProjectFile);
-            
+
             VersionVariables versionVariables;
             if (!VersionAndBranchFinder.TryGetVersion(SolutionDirectory, out versionVariables, NoFetch, new Authentication(), fileSystem))
             {
@@ -80,7 +80,6 @@
 
         void CreateTempAssemblyInfo(VersionVariables versionVariables)
         {
-
             if (IntermediateOutputPath == null)
             {
                 var tempFileName = string.Format("AssemblyInfo_{0}_{1}.g.cs", Path.GetFileNameWithoutExtension(ProjectFile), Path.GetRandomFileName());
@@ -92,7 +91,7 @@
             }
 
             var assemblyInfoBuilder = new AssemblyInfoBuilder();
-            var assemblyInfo = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables);
+            var assemblyInfo = assemblyInfoBuilder.GetAssemblyInfoText(versionVariables, Path.GetFileNameWithoutExtension(ProjectFile));
             File.WriteAllText(AssemblyInfoTempFilePath, assemblyInfo);
         }
     }

